sdk/python/lib/pulumi/runtime/stack.py

```diff
@@ -174,54 +174,65 @@ def massage(attr: Any, seen: List[Any]):
     if is_primitive(attr):
         return attr
 
+    if isinstance(attr, Output):
+        return attr.apply(lambda v: massage(v, seen))
+
+    if isawaitable(attr):
+        return Output.from_input(attr).apply(lambda v: massage(v, seen))
+
     # from this point on, we have complex objects.  If we see them again, we don't want to emit them
     # again fully or else we'd loop infinitely.
     if reference_contains(attr, seen):
-        # Note: for Resources we hit again, emit their urn so cycles can be easily understood in
```

Comment:
Why remove the comment?

Reply:
Good catch, we can probably leave this in here.